8 header of all pages

Eric-ZheranLi opened the pull request:
1. Create src\components\Header folder, and implement header design.
2. Update header component in src\layout, and update and move Header.test.tsx from __tests__\layouts to __tests__\components

tatoMa commented on src/components/Header/AccountLabel.js:
```diff
@@ -0,0 +1,13 @@
+  return (
+    <div className={styles.account_label}>
+      <div className={styles.account_label__user_name_div}>
+        <div className={styles.account_label__first_letter}>{userName.charAt(0).toUpperCase()}</div>
```
When you have something to compute, extract it into a function/method, so it can be reusable and have a better file structure.

tatoMa commented on src/components/Header/Header.module.scss:
```diff
@@ -0,0 +1,107 @@
+    justify-content: center;
+    align-items: center;
+    padding: 24px 370px;
+
```
No empty line is needed anywhere in this file.

Eric-ZheranLi replied:
I have checked several times and tried to eliminate all unnecessary empty line :)

tatoMa commented on src/components/Header/Header.module.scss:
```diff
@@ -0,0 +1,107 @@
+    width: 1920px;
+    height: 88px;
+    
+    /* Primaty/Grey Light */
+    background: #F6F5F4;
```
If this color refers from our design system(theme CSS), what if the theme CSS file changes the color value of grey light?

Eric-ZheranLi replied:
https://github.com/taodemy/taotalk/blob/457af72abe43f54f6b3c079e56f580a15ec67f7f/src/components/Header/Header.module.scss#L1
https://github.com/taodemy/taotalk/blob/457af72abe43f54f6b3c079e56f580a15ec67f7f/src/components/Header/Header.module.scss#L9-L10
https://github.com/taodemy/taotalk/blob/457af72abe43f54f6b3c079e56f580a15ec67f7f/src/components/Header/Header.module.scss#L33-L34

I have imported color settings from the design system, and replace color in scss file from hard coding to imported variables.

tatoMa commented:
Do you have a similar error when you run the dev server?

<img width="1542" alt="image" src="https://github.com/taodemy/taotalk/assets/12379895/9a692158-444b-400f-a0eb-56de3d3e7255">

Double-Twelve commented on src/components/Header/Header.module.scss:
```diff
@@ -0,0 +1,107 @@
+.header {
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+    align-items: center;
+    padding: 24px 370px;
+
+    position: relative;
+    width: 1920px;
+    height: 88px;
```
1. don't need to set `flex-direction: column;` `align-items: center;`
2. don't need to set `height: 88px;`, and change padding to `padding: 24px 0px;`
3. don't need to set `width: 1920px;`

Eric-ZheranLi replied:
https://github.com/taodemy/taotalk/blob/457af72abe43f54f6b3c079e56f580a15ec67f7f/src/components/Header/Header.module.scss#L3-L11

Previous codes are modified as suggestions in guidance. And other similar problems in css file are also modified accordingly.

Double-Twelve commented on src/components/Header/Header.module.scss:
```diff
@@ -0,0 +1,107 @@
+  }
+  
+.content {
+    display: flex;
+    flex-direction: row;
+    justify-content: space-between;
+    align-items: center;
+
+    width: 1180px;
+    height: 40px;
```
1. don't need to set `flex-direction: row;`, `width: 1180px;`, `height: 40px;`
2. add `min-width: 1180px;`

Eric-ZheranLi replied:
https://github.com/taodemy/taotalk/blob/457af72abe43f54f6b3c079e56f580a15ec67f7f/src/components/Header/Header.module.scss#L13-L18

Previous codes are modified as suggestions in guidance. And other similar problems in css file are also modified accordingly.

Eric-ZheranLi commented:
I have done following modification in the recent commit:

1. use semantic elements
I have replaced most of unnecessary <div> tags in Header/index.tsx to semantic elements.

2.  rename component as LogInButton
I changed the components name of AccountLabel to LogInButton for a better understanding, and also for the log out status.

3.  delete Title component
I use <ul>, <li>, and <a> to print all nav links instead, and delete the previous Title component.

@tatoMa @Double-Twelve  Please help me check again the files. Sorry for spending so much time on this issue.

tatoMa commented:
> I have done following modification in the recent commit:
> 
> 1. use semantic elements
>    I have replaced most of unnecessary <div> tags in Header/index.tsx to semantic elements.
> 2. rename component as LogInButton
>    I changed the components name of AccountLabel to LogInButton for a better understanding, and also for the log out status.
> 3. delete Title component
>    I use <ul>, <li>, and <a> to print all nav links instead, and delete the previous Title component.
> 
> @tatoMa @Double-Twelve Please help me check again the files. Sorry for spending so much time on this issue.

So you did not notice the font family looks so different between our UI design and your PR?
<img width="1720" alt="image" src="https://github.com/taodemy/taotalk/assets/12379895/0fdd7bdd-a12f-45e2-a8aa-3d20fa7d53d6">

Eric-ZheranLi commented:
> > I have done following modification in the recent commit:
> > 
> > 1. use semantic elements
> >    I have replaced most of unnecessary  tags in Header/index.tsx to semantic elements.
> > 2. rename component as LogInButton
> >    I changed the components name of AccountLabel to LogInButton for a better understanding, and also for the log out status.
> > 3. delete Title component
> >    I use , * , and  to print all nav links instead, and delete the previous Title component.
> > @tatoMa @Double-Twelve Please help me check again the files. Sorry for spending so much time on this issue.
> 
> So you did not notice the font family looks so different between our UI design and your PR? <img alt="image" width="1720" src="https://user-images.githubusercontent.com/12379895/244986557-0fdd7bdd-a12f-45e2-a8aa-3d20fa7d53d6.png">

---------------------------------------------------------------------------------------

I have added necessary font import from google fonts to global.scss file to make the font family look same as in figma. And also fix some minor display issues in header file.
